department-of-veterans-affairs/caseflow
# Enable CAVC MDR appeal stream creation

Resolves [CASEFLOW-92](https://vajira.max.gov/browse/CASEFLOW-92)

### Description
Update the CAVC controller and model to allow for the creation of the MDR subtype.

### Acceptance Criteria
- [x] The "Add a Court Remand" form on dispatched appeals now allows for the creation of MDR CAVC remand appeal streams.
- [x] This feature should be accessible to the following user groups: CAVC Litigation Support
Include screenshot(s) in the Github issue if there are front-end changes
- [ ] Update documentation: CAVC Remand
- [x] Form does not validate Mandate & Judgement dates for MDR flow
- [x] Verify a new appeal stream is created when the CAVC Litigation Support team selects MDR and clicks submit on the CAVC entry form
- [x] Verify the new appeal stream is linked to the existing docket number at the Board
- [x] Controller does not require mandate and judgement dates (model will do validation)
model kicks off a new appeal stream even if the mandate and judge dates are empty, but only if an mdr remand
- [x] The new appeal's Root Task, Distribution Task, & CAVC Task are created
- [x] Verify the original and new appeal appears when searching for original docket #
- [x] Does not include any work around the MDR Task, that is Caseflow-93
- [x] No smart date disabling on the form for MDR.

### Testing Plan
1. Open up your console and create a ready for CAVC appeal, note the `stream_docket_number` so you can search for the appeal in the UI.
```ruby
original_appeal = FactoryBot.create(:appeal, 
                           :dispatched, 
                           associated_judge: JudgeTeam.first.judge, 
                           associated_attorney: JudgeTeam.first.attorneys.first)
FactoryBot.create_list(:request_issue, 2,                                                                                  
            :rating,                                                                                            
            :with_rating_decision_issue,                                                                           
            decision_review: original_appeal,                                                                               
            veteran_participant_id: Veteran.first.participant_id,                                                        
            contested_issue_description: "issue_description",                                                              
            notes: "issue_notes")
```
2. Login as a CAVC lit support user
3. Search for the dispatched appeal you just created and select it
4. Click the "+ Add CAVC Remand" button in the "Post-Despatch actions" section
5. Fill out the "Add a Court Remand" form, note that an MDR remand does not require entry of a judgement or mandate date however they may be entered if they do exist
6. Open up the Network tab of your Chrome devtools and Submit the form
7.  Check that you received a 201 response from the backend (`appeals/<uuid>/cavc_remand`) confirming the creation of the MDR stream, also note the success banner in the UI confirming a MDR creation.
8. While still in the network tab, open the preview tab (if not already open) and note the `id` on the `cavc_appeal`
9. Open your console once more and search for the newly created appeal with the `id` you just grabbed
10. Check that the appeal was correctly created with the `DistributionTask`, `CavcTask` and `SendCavcRemandProcessedLetterTask` tasks
11. Navigate to http://localhost:3000/search and search for the docket number of your original appeal and confirm you can see the newly created MDR appeal stream

### Code Documentation Updates
- [ ] CAVC Remand wiki already [references](https://github.com/department-of-veterans-affairs/caseflow/wiki/CAVC-Remands#supported-remand-subtypes) the support of MDR remands
